src/metaxy/graph/status.py

```diff
@@ -0,0 +1,245 @@
+"""Feature metadata status inspection utilities.
+
+This module provides reusable SDK functions for inspecting feature metadata status,
+useful for both CLI commands and programmatic usage.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Any
+
+import narwhals as nw
+import polars as pl
+from pydantic import BaseModel, Field
+
+from metaxy.models.types import FeatureKey
+
+if TYPE_CHECKING:
+    from metaxy.metadata_store.base import MetadataStore
+    from metaxy.models.feature import BaseFeature
+    from metaxy.versioning.types import LazyIncrement
+
+
+class FeatureMetadataStatus(BaseModel):
+    """Status information for feature metadata in a metadata store.
+
+    This model encapsulates the current state of metadata for a feature,
+    including whether it exists, needs updates, and sample counts.
+    """
+
+    feature_key: FeatureKey = Field(description="The feature key being inspected")
+    target_version: str = Field(description="The feature version from code")
+    metadata_exists: bool = Field(description="Whether metadata exists in the store")
+    row_count: int = Field(description="Number of metadata rows (0 if none exist)")
+    added_count: int = Field(description="Number of samples that would be added")
+    changed_count: int = Field(description="Number of samples that would be changed")
+    needs_update: bool = Field(description="Whether updates are needed")
+    lazy_increment: LazyIncrement | None = Field(
```

Comment:
This should not be on the same model. Use a NamedTuple or smth. 
Let's keep Pydantic models pure from arbitrary types. It will be useful later.